### DISCUSSION TITLE
Issue opening AR on android if more than one model present on same page

### DISCUSSION THREAD

**Discussion by wearept** [google/model-viewer#2330] [Q&A]
Hi,

We are a bit confused and uncertain about how to troubleshoot this issue if somebody could point us in the direction on how to solve the issues we would be grateful!

At https://artexperiences.pt there are several artist pages with AR content that until last Friday was working perfectly but since yesterday they cause the ar mode to present an error and eventually hang and crash. 

Each page like : https://www.artexperience.pt/#/artists/teresarego has a top AR that is still working perfectly and 4 individual models that now present this issue. the page code hasn't changed but since then we have been trying to change positioning and other things but with no success figuring out what can be done to stabilize this situation.

We tried changing the order and ar-placements but without success. Again, this was working without any issues before starting changes trying to figure out what is happening. 

We also tried not linking to the actual script version but to previous ones, and even downloading the js and pointing to our servers but, still the same issue was present.

This is happening in different androids from our team. No issues with ios, no issues previewing objects on browsers, just crashing on ar.

We tend to think that something changed maybe with andoid updates? 
If someone is also having such issues can you please provide feedback?

Thank you!
Eduardo

**Comment by wearept** [comment]
We changed all ar-placement to "floor" and the issue is bypassed.
We will maintain temporarily this link with main experience placed at floor and individual ones placed at wall if it is somehow helpful for debugging... https://artexperience.pt/#/artists/fatimabravo_

**Reply by elalish** [reply]
Yes, I presume you were using scene-viewer as your AR mode? Being a separate app, it gets updated independently and I believe you hit a recent bug. I also believe it has already been fixed, so hopefully you will get an update shortly to improve things. 

As part of our on-going poll: why do you prefer scene-viewer to our default WebXR mode?

**Reply by wearept** [reply]
Hi Elalish, thank you for the feedback! Our preference for the scene-viewer is that we like to create experiences that enable people to not just have a different perception of 3D elements but also to interact.  One of the interaction modalities being the ability to take pictures or create videos to share with friends and webXR doesn't support that, or if there is a workaround to make it work we are still to discover. :) 

Again, thank you not just for this support but for all the great effort you have been doing to make this become something really awesome for all!

**Reply by david-rhodes** [reply]
@elalish sorry to jump in here, but we've been experiencing a wide range of bugs with AR models not loading in scene-viewer in the last few weeks. I see Google Play Services for AR was updated on my device on April 15th. I don't recall seeing these issues before that. Do you think this could be related? If so, do you have any more information you can point me to regarding resolution, update ETA, etc.? 

Also, in general, is there a better way to file scene-viewer specific bugs? Thank you and apologies for hijacking this thread if it's not related.

**Reply by elalish** [reply]
@wearept Thanks for the feedback; we are indeed working to bring the recording feature to WebXR mode. Stay tuned!

@david-rhodes Yes, definitely related. There should be a fixed version out now; have you seen a more recent update? If anything is still broken, please let us know. We don't have a better place to file bugs than here.

**Reply by david-rhodes** [reply]
@elalish is the fix a part of Google Play Services for AR, or the Google App itself?

**Reply by elalish** [reply]
Scene Viewer is part of the Google App, while ARCore is part of Google Play Services for AR. This bug is in the first one.

**Reply by david-rhodes** [reply]
Thank you for the clarification!
